packages/binary/messagelayer/messageparser/message_parser.go

```diff
@@ -28,17 +27,7 @@ func New() (result *MessageParser) {
 	result = &MessageParser{
 		bytesFilters:   make([]BytesFilter, 0),
 		messageFilters: make([]MessageFilter, 0),
-		Events: Events{
-			MessageParsed: events.NewEvent(func(handler interface{}, params ...interface{}) {
-				handler.(func(*message.Message, *peer.Peer))(params[0].(*message.Message), params[1].(*peer.Peer))
-			}),
-			BytesRejected: events.NewEvent(func(handler interface{}, params ...interface{}) {
-				handler.(func([]byte, error, *peer.Peer))(params[0].([]byte), params[1].(error), params[2].(*peer.Peer))
-			}),
-			MessageRejected: events.NewEvent(func(handler interface{}, params ...interface{}) {
-				handler.(func(*message.Message, error, *peer.Peer))(params[0].(*message.Message), params[1].(error), params[2].(*peer.Peer))
-			}),
-		},
+		Events:         *newEvents(),
```

Comment:
Looking at the `DRNG` or `ValueTangle` the `Events` are a pointer. Maybe that should be the case here, too?